csrc/host_ir/executor.cpp

```diff
@@ -222,6 +222,30 @@ HostIrEvaluator::HostIrEvaluator(
       "Inputs cannot be aliased");
 }
 
+KernelArgumentHolder HostIrEvaluator::runWithInputs(
```

Comment:
Do you expect that eventually we'll remove the other runWithInputs?

Reply:
I'll need to understand cache ID more to tell. 

Currently, the KernelArgumentHolder version is added for performance. So it'll likely exist before I can make the other version (which takes a map) equally fast. 

The map version is more user friendly because it explicitly says which Val* to bind. This is remotely analogous to positional arguments vs keyword arguments.